src/main/java/tech/fastj/systems/audio/AudioEventListener.java

```diff
@@ -34,6 +36,9 @@
                         audioEventListener.audioStartAction.accept(audioEvent);
                         break;
                     }
+                    default: {
+                        throw new IllegalStateException("audio state was unexpected and invalid");
```

Comment:
For both this segment and the other segment, it would be a good idea to include the playback state that caused the exception.

Reply:
Just to make sure, do you mean to include `audioEventListener.audio.getPreviousPlaybackState()` as a string in the exception call, and so on for the second segment?

Reply:
Yep, just as a string in the exception(s). You're also free to abstract that method call into a value -- either or is fine with me.